WebRoot/js/zimbraMail/share/view/dialog/ZmUploadDialog.js

```diff
@@ -479,28 +477,37 @@ ZmUploadDialog.prototype._addFileInputRow = function(oneInputOnly) {
     }
 };
 
-ZmUploadDialog.prototype._handleFileSize =
-function(inputEl, sizeEl){
+ZmUploadDialog.prototype._handleFileSelection =
+function(inputEl, fileDetailEl){
 
     var files = inputEl.files;
-    if(!files) return;
+    if(!files || !files.length){
+        fileDetailEl && (fileDetailEl.innerHTML = "");
+        return;
+    }
 
-    var sizeStr = [], className, totalSize =0;
+    var sizeStr = [], className, fileNameDetail, totalSize =0;
     for(var i=0; i<files.length;i++){
         var file = files[i];
         var size = file.size || file.fileSize /*Safari*/ || 0;
-	    var aCtxt = ZmAppCtxt.handleWindowOpener();
+        var aCtxt = ZmAppCtxt.handleWindowOpener();
         if(size > aCtxt.get(ZmSetting.DOCUMENT_SIZE_LIMIT))
             className = "RedC";
         totalSize += size;
     }
 
-    if(sizeEl) {
-        sizeEl.innerHTML = "  ("+AjxUtil.formatSize(totalSize, true)+")";
+    if(files.length > 1) {
```

Comment:
I don't think there's any gain with that. The files object is in scope and so does the property length on that. It would be unnecessary creating another variable here.